File: FinalProj/GCD.cpp
#include <stdlib.h>
#include <iostream>

using namespace std;

void Euclidean_GCD(int x, int y);
int NoName_GCD(int x, int y);

int main()
{
  int num_one, num_two;

  //cout << "Enter the first number: ";
  //cin >> num_one;

  //cout << "Enter the second number: ";
  //cin >> num_two;

  Euclidean_GCD(45, 15);
  NoName_GCD(45, 15);
  return 0;
}


void Euclidean_GCD(int x, int y)
{
  if (y > 0)
  {
    int temp_x = x;
    x = y;
    y = temp_x % y;
    Euclidean_GCD(x, y);
  }
  else
    cout << "Euclidean_GCD is: " << x << endl;
}

int NoName_GCD(int x, int y)
{
  if (y > x)
  {
    int temp = y;
    y = x;
    x = temp;
  }

  if (x == y)
    cout << "NoName_GCD is: " << x << endl;
  else if ((x%2) == 0 && (y%2) == 0)
    return (2 * NoName_GCD((x/2), (y/2)));
  else if ((x%2) != 0 && (y%2) == 0)
    return (NoName_GCD(x, (y/2)));
  else if ((x%2) == 0 && (y%2) != 0)
    return (NoName_GCD((x/2), y));
  else if ((x%2) != 0 && (y%2) != 0)
    return (NoName_GCD((x-y), y));

}

File: FinalProj/GCD.py
def main():
    x = Euclidean_GCD(40, 15)
    print("Euclidean_GCD is: ", x)
    y = NoName_GCD(40, 15)
    print("NoName_GCD is: ", y)

def Euclidean_GCD(x, y):
    if y > 0:
        temp_x = x;
        x = y;
        y = temp_x % y
        return Euclidean_GCD(x, y)
    else:
        return x
    
def NoName_GCD(x, y):
    if (y > x):
        temp = y
        y = x
        x = temp
        
    if (x == y):
        return x
    
    elif ((x%2) == 0 and (y%2) == 0):
        return (2 * NoName_GCD((x/2), (y/2)))
    
    elif ((x%2) != 0 and (y%2) == 0):
        return (NoName_GCD(x, (y/2)))
            
    elif ((x%2) == 0 and (y%2) != 0):
        return (NoName_GCD((x/2), y))
    
    elif ((x%2) != 0 and (y%2) != 0):
        return (NoName_GCD((x-y), y))   
    

main()